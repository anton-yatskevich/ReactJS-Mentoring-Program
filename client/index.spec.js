import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AppRender from './index';

describe('Root component test', () => {
    let componentSnapshot;

    beforeEach(() => {
        ReactDOM.render = jest.fn((component) => {
            componentSnapshot = shallow(component);
        });
    });

    it('should call ReactDOM.render and match snapshot', () => {
        AppRender();

        expect(ReactDOM.render).toHaveBeenCalled();
        expect(toJson(componentSnapshot)).toMatchSnapshot();
    });
});
