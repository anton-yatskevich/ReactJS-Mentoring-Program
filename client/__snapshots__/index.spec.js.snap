// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root component test should call ReactDOM.render and match snapshot 1`] = `<AppContainer />`;
